File: Labs/Lab01-Java/lab01Rec.java
import java.util.Scanner;

public class lab01Rec {

    public static int maiusculo(String palavra, int i){
        
       if(i == palavra.length()){ // Caso base
            return 0;

       }else if(palavra.charAt(i) >= 65 && palavra.charAt(i) <= 90){
            return 1 + maiusculo(palavra, i + 1); //Chamada recursiva

       }else{
            return maiusculo(palavra, i + 1); //Chamada recursiva
       }

    }


    public static void main(String[] args) {
        
        try (Scanner scanner = new Scanner(System.in)) {

            System.out.println("Digite a palavra");
            String palavra = scanner.nextLine();

            while (!palavra.equals("FIM")){ //Equals compara o conteudo de dois objetos

                int resul = maiusculo(palavra,0);

                System.out.println("A palavra possui " + resul + " letras MAIUSCULAS");

                System.out.println("Digite a palavra");
                palavra = scanner.nextLine();
            }

            scanner.close();
        }
    }
    
}

File: Labs/Lab01-C/lab01Rec.c
#include <stdio.h>


    int maiusculo(char palavra[], int i){

        if(palavra[i] == '\0'){ // Caso Base
            return 0;

        }else if(palavra[i] >= 65 && palavra[i] <= 90){ 
            return 1 + maiusculo(palavra, i + 1);// Chamada Recursiva

        }else{
            return maiusculo(palavra, i + 1); // Chamada Recursiva
        }
    }
    

    int main(){

        printf("Digite a palavra: ");
        char palavra[100];
        scanf("%s", palavra);

        while (palavra[0] != 'F' && palavra[1] != 'I' && palavra[2] != 'M' && palavra[3] != '\0'){ 
        
            int resp = maiusculo(palavra, 0);

            printf("A palavra possui %d letras maiusculas", resp);

            printf("\n");

            printf("Digite a palavra: ");
            scanf("%s", palavra);

        }
             

        return 0;
        
    }
    

